builtin/providers/terraform/provider_test.go

```diff
@@ -3,28 +3,27 @@ package terraform
 import (
 	"testing"
 
-	"github.com/hashicorp/terraform/helper/schema"
-	"github.com/hashicorp/terraform/terraform"
+	"github.com/hashicorp/terraform/providers"
 )
 
-var testAccProviders map[string]terraform.ResourceProvider
-var testAccProvider *schema.Provider
+var testAccProviders map[string]*Provider
+var testAccProvider *Provider
 
 func init() {
-	testAccProvider = Provider().(*schema.Provider)
-	testAccProviders = map[string]terraform.ResourceProvider{
+	testAccProvider = NewProvider()
+	testAccProviders = map[string]*Provider{
 		"terraform": testAccProvider,
 	}
 }
 
-func TestProvider(t *testing.T) {
-	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
-		t.Fatalf("err: %s", err)
-	}
-}
+// func TestProvider(t *testing.T) {
+// 	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
+// 		t.Fatalf("err: %s", err)
+// 	}
+// }
```

Comment:
`configschema.Block` has its own version of `InternalValidate`, so if you split the data source schema out into a separate function as I'd suggested in another comment it could be nice to internal-validate its schema here, which would achieve something similar to what this function was doing before.

`InternalValidate` is a way for us to make sure the schema is properly constructed, so it deals with things like making sure every attribute has either `Required`, `Optional`, or `Computed` set, that they aren't set in combinations that make no sense, that `MaxItems` and `MinItems` are used correctly on blocks, etc.

Reply:
I added an `InternalValidate()` test for the data source schema (in data_source_state_test.go). Should there also be a provider-level schema internal-validate as well?

Reply:
Since this provider doesn't have any configuration of its own, I suppose there isn't anything to `InternalValidate` at the provider level right now. Sorry for not realizing that on the first pass through here.